pennylane/compiler/qjit_api.py

```diff
@@ -96,6 +97,22 @@ def circuit(theta):
     >>> circuit(0.5)  # the precompiled quantum function is called
     array(0.)
 
+    :func:`~.qjit` compiled programs also support nested container types as inputs and outputs of
```

Comment:
@josh146 @timmysilv Are you guys happy with the addition of this example to describe the PyTree support in Catalyst? (related to our thread here https://github.com/PennyLaneAI/pennylane/pull/4698/files#r1409940603)

Reply:
@josh146 I'll add your example there instead [5b89179](https://github.com/PennyLaneAI/pennylane/pull/4698/commits/5b891799f4a502cc30f3bee17e5d999d50fbee9c)

Reply:
I'm fixing the issue with the docs build failure..

Reply:
👍